### PR TITLE
Base UI setup for sample details refactor

### PR DESCRIPTION
**Pull request by deepsidhu85**
## Description of changes
What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.  Include how to test your changes.

Updated SampleDetailViewer tabs to support sample details refactor into this viewer (just base UI work and sample/file data changes)

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
~* [ ] Tests added (or description of how to test) for any new features.~
~* [ ] User documentation updated for UI or technical changes.~

### CODE REVIEW

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/ajax/dto/SampleSequencingObjectFileModel.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package ca.corefacility.bioinformatics.irida.ria.web.ajax.dto;
+
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequencingObject;
```
Comment (joshsadam) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/ajax/dto/SampleSequencingObjectFileModel.java:3]: I thought we were going to move the ajax and dto's within the directory of their respective controller?

Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/ajax/dto/SampleSequencingObjectFileModel.java:3]: Yup that's the plan :)

**Comment by joshsadam** [comment]
I really like the way this is looking, going to go through a couple more things.  Can you see why you tests are failing.

File: src/main/webapp/resources/js/pages/launch/LaunchFiles.jsx [hunk 2]
```diff
@@ -60,7 +60,7 @@ export function LaunchFiles() {
         const firstSamples = [];
         data.forEach((sample) => {
           if (sample.files.length) {
-            sample.selected = sample.files[0].identifier;
+            sample.selected = sample.files[0].fileInfo.identifier;
```
Comment (joshsadam) [inline src/main/webapp/resources/js/pages/launch/LaunchFiles.jsx:63]: Is the plan to move this into redux tool-kit?  This might be it's own API.

File: src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx [hunk 3]
```diff
@@ -9,26 +21,79 @@ import { Empty, List } from "antd";
...
-                {metadata[item].value}
-              </span>
-            }
+  const [visible, setVisible] = React.useState(false);
+  return (
+    <>
+      <Button onClick={() => setVisible(true)}>
+        {i18n("SampleMetadata.addNewMetadata")}
+      </Button>
+      <Modal
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx:30]: Make this it's own component.  Use the sample coding style that I use for the root component of the sample details viewer to allow different types of buttons to open the modal.

File: src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx [hunk 4]
```diff
@@ -9,26 +21,79 @@ import { Empty, List } from "antd";
...
+            label={i18n("SampleMetadata.modal.permission")}
+          >
+            <Select
+              defaultValue="1"
+              style={{ width: "100%" }}
+              onChange={(e) =>
+                console.log("Changed to LEVEL " + e + " metadata permission.")
+              }
+            >
+              <Option value="1">{i18n("SampleMetadata.modal.level1")}</Option>
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx:61]: Do you think we should make getting these an ajax request?  That way if they change we only need to update the server and not different places in the UI?

File: src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx [hunk 5]
```diff
@@ -9,26 +21,79 @@ import { Empty, List } from "antd";
...
+        {Object.keys(metadata).length ? (
+          <List
+            itemLayout="horizontal"
+            dataSource={Object.keys(metadata).sort((a, b) =>
+              a.localeCompare(b)
+            )}
+            renderItem={(item) => (
+              <List.Item className="t-sample-details-metadata-item">
+                <List.Item.Meta
+                  title={
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx:79]: We should consider making theses deletable - it seems to be a common request.

File: src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx [hunk 6]
```diff
@@ -9,26 +21,79 @@ import { Empty, List } from "antd";
...
+        {i18n("SampleMetadata.addNewMetadata")}
+      </Button>
+      <Modal
+        className="t-add-metadata-field"
+        onCancel={() => setVisible(false)}
+        visible={visible}
+        onOk={() => setVisible(false)}
+      >
+        <Title level={4}>{i18n("SampleMetadata.modal.title")}</Title>
+        <Form>
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx:37]: Can you make the layout on this vertical so that the labels are on top of the inputs (like every single other form on this project ;)).

File: src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx [hunk 7]
```diff
@@ -9,26 +21,79 @@ import { Empty, List } from "antd";
...
+      <div>
+        {Object.keys(metadata).length ? (
+          <List
+            itemLayout="horizontal"
+            dataSource={Object.keys(metadata).sort((a, b) =>
+              a.localeCompare(b)
+            )}
+            renderItem={(item) => (
+              <List.Item className="t-sample-details-metadata-item">
+                <List.Item.Meta
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx:78]: Because this list of metadata fields could get seriously long.  Can you put this into a scrollable container (this can be done later if you want).

**Review by joshsadam — CHANGES_REQUESTED** [review]
Some questions and some changes.  Let me know if you have any questions.

File: src/main/webapp/resources/js/pages/launch/LaunchFiles.jsx (hunk 2 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/launch/LaunchFiles.jsx:63]: Yup that's the plan in a future PR for this refactor

File: src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx (hunk 3 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx:30]: Sounds good! Thanks

File: src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx (hunk 4 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx:61]: Yup that's the plan :)

File: src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx (hunk 5 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx:79]: Yeah we will need to discuss this when I work on this component

File: src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx (hunk 6 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx:37]: Sounds good

File: src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx (hunk 7 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx:78]: Sure thing :)
